_posts/2023-09-27-onstove-workflow.md

```diff
@@ -0,0 +1,261 @@
+of this script is a ``model.pkl`` file containing all the information of our model.
+
+```python
+rule prepare_model:
+   input:
+        prep_file = "Prep_files/{country}_prep_file.csv",
+        techs_file = r"Technical_specs\{scenario}\{country}_file_tech_specs.csv",
+        mask_layer = rules.process_data.output.mask_layer,
+        population = rules.process_data.output.population,
+        ghs = rules.process_data.output.ghs,
+        forest = rules.process_data.output.forest,
+        biomass_friction = rules.process_data.output.biomass_friction,
+        mv_lines = rules.process_data.output.mv_lines,
+        ntl = rules.process_data.output.ntl,
+        traveltime_cities = rules.process_data.output.traveltime_cities,
+        roads = rules.process_data.output.roads,
+        temperature = rules.process_data.output.temperature,
+        water = rules.process_data.output.water,
+        buffaloes = rules.process_data.output.buffaloes,
+        cattles = rules.process_data.output.buffaloes,
+        poultry = rules.process_data.output.poultry,
+        goats = rules.process_data.output.goats,
+        pigs = rules.process_data.output.pigs,
+        sheeps = rules.process_data.output.sheeps
+   params:
+         wealth_index = r"GIS-data\Poverty\{country}_relative_wealth_index",
+         output_directory = "Results/{country}/{scenario}",
+         country = "{country}"
+   output:
+         model = "Results/{country}/{scenario}/model.pkl"
+   script:
+         "scripts/model_preparation.py"
+```
+
+### The `run_model` rule <a name="run_model"></a>
+Next, we define a ``run_model`` rule, where we read in the prepared model file ``model.pkl`` 
+and additional scenario data for each specific country. The output expected is a ``results.pkl`` 
+file for each country, which are produced by the ``model_run.py`` script.
+
+```python
+rule run_model:
+   input:
+        model = rules.prepare_model.output.model,
+        scenario_file = r"Results\Scenario_files\{scenario}\{country}_scenario_file.csv"
+   params:
+         output_directory = "Results/{country}/{scenario}/{restriction}",
+         country = "{country}",
+         restriction = "{restriction}"
+   output:
+         results = "Results/{country}/{scenario}/{restriction}/results.pkl",
+   script:
+        "scripts/model_run.py"
+```
+
+### The `merge_results` rule <a name="merge_results"></a>
+Finally, after having all result files for all countries, the ``merge_results`` rule takes care 
+of reading all countries results and merging them into a large Africa model. The expected 
+``output`` is a ``results.pkl`` file for Africa.
+
+```python
+rule merge_results:
+   input:
+        results = expand("Results/{country}/{{scenario}}/{{restriction}}/results.pkl",
+                         country=COUNTRIES),
+        boundaries = r"GIS-data\Admin\Admin_1.shp"
+   params:
+         output_directory = "Results/Africa/{scenario}/{restriction}",
```

Comment:
Note you are hard-coding in the continent name here. What if you wanted to perform the analysis for South East Asia?

Reply:
This is a great point, I will definitely consider making the workflow more general as currently (as you said) it has some hardcoded parts specific to the Africa analysis.